tests/devices/test_default_mixed.py

```diff
@@ -488,14 +485,43 @@ def test_channel_root(self, x, tol):
         root = np.reshape(root_state(nr_wires), [2] * 2 * nr_wires)
         dev._state = root
         kraus = dev._get_kraus(op)
-        if (
-            op.name == "CNOT"
-            or op.name == "ISWAP"
-            or (isinstance(op, PauliError) and nr_wires == 2)
-        ):
-            dev._apply_channel(kraus, wires=Wires([0, 1]))
-        else:
-            dev._apply_channel(kraus, wires=Wires(0))
+        getattr(dev, apply_method)(kraus, wires=op.wires)
+        assert np.allclose(dev._state, target_state, atol=tol, rtol=0)
+
+    ops = [
+        PauliX(wires=0),
+        PauliX(wires=2),
+        Hadamard(wires=0),
+        CNOT(wires=[0, 1]),
+        ISWAP(wires=[0, 1]),
+        SWAP(wires=[2, 0]),
+        MultiControlledX(wires=[0, 1, 2]),
+        MultiControlledX(wires=[2, 0, 1]),
+        AmplitudeDamping(0.5, wires=0),
+        DepolarizingChannel(0.5, wires=0),
+        ResetError(0.1, 0.5, wires=0),
+        PauliError("Z", 0.3, wires=0),
+        PauliError("XY", 0.5, wires=[0, 1]),
+        PauliError("XZY", 0.1, wires=[0, 2, 1]),
+    ]
+
+    @pytest.mark.parametrize("op", ops)
+    @pytest.mark.parametrize("num_dev_wires", [1, 2, 3])
+    def test_channel_against_matmul(self, num_dev_wires, op, apply_method, tol):
+        if num_dev_wires < max(op.wires) + 1:
+            pytest.skip("Need at least as many wires in the device as in the operation.")
```

Comment:
If a error is emitted I would catch it instead of skipping the test.

Reply:
Sorry, I'm not sure what you mean? The skipped tests are "nonsensical" tests, like testing a `CNOT` on a single-qubit device, which can't be executed.